flink-streaming-connectors/flink-connector-activemq/src/main/java/org/apache/flink/streaming/connectors/activemq/internal/AMQExceptionListener.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.streaming.connectors.activemq.internal;
+
+import org.slf4j.Logger;
+
+import javax.jms.ExceptionListener;
+import javax.jms.JMSException;
+
+public class AMQExceptionListener implements ExceptionListener {
+
+	private final boolean logFailuresOnly;
+	private final Logger logger;
+
+	public AMQExceptionListener(Logger logger, boolean logFailuresOnly) {
+		this.logger = logger;
+		this.logFailuresOnly = logFailuresOnly;
+	}
+
+
+	@Override
+	public void onException(JMSException e) {
+		if (logFailuresOnly) {
+			logger.error("Received ActiveMQ exception", e);
+		} else {
+			throw new RuntimeException("Received ActiveMQ exception", e);
```

Comment:
I don't think that this works as expected. The problem is that the exception is thrown somewhere in the activemq code then, not stopping the Flink task.
What you need to do is use some field in the source to report an async exception. Check for example the `FlinkKafkaProducerBase` how its done.